### PR TITLE
JR2-954 Add token check

### PR DESCRIPTION
**Pull request by Invizible**
related https://github.com/alv-ch/job-ad-service/pull/105

Allows to view and cancel 'ROLE_PRIVATE_EMPLOYMENT_AGENT' or 'ROLE_PUBLIC_EMPLOYMENT_SERVICE' users or anonymous and other users with a valid token

Bear in mind anonymous user can still view a jobAd by updating the route from 'job-publication-detail' to 'job-detail' but not cancel it

### CODE REVIEW

File: src/main/webapp/app/shared/job-advertisement/job-advertisement.service.ts [hunk 1]
```diff
@@ -59,6 +63,13 @@ export class JobAdvertisementService {
             .map((result) => result.status);
     }
 
+    cancelByToken(jobAdCancelRequest: JobAdvertisementCancelRequest): Observable<number> {
```
Comment (cjpsycos) [inline src/main/webapp/app/shared/job-advertisement/job-advertisement.service.ts]: This should go to /jobadservice/api/jobAdvertisements/{id}/cancel?token={accessToken}

File: src/main/webapp/app/shared/auth/auth-jwt.service.ts [hunk 2]
```diff
@@ -20,7 +20,7 @@ export class AuthServerProvider {
             password: credentials.password,
             rememberMe: credentials.rememberMe
         };
-        return this.http.post(SERVER_API_URL + 'api/authenticate', data, {observe: 'response'}).map(authenticateSuccess.bind(this));
+        return this.http.post(SERVER_API_URL + 'api/authenticate', data, { observe: 'response' }).map(authenticateSuccess.bind(this));
```
Comment (cjpsycos) [inline src/main/webapp/app/shared/auth/auth-jwt.service.ts]: Does this work with eiam?

Comment (Invizible) [reply src/main/webapp/app/shared/auth/auth-jwt.service.ts]: I do not have answer to your question, I just fixed styling in order to make build success

File: src/main/webapp/app/dashboard/job-publication-detail/job-publication-detail.component.ts [hunk 3]
```diff
@@ -29,11 +30,16 @@ export class JobPublicationDetailComponent {
...
+    token: string;
 
     constructor(private jobAdvertisementService: JobAdvertisementService,
                 private store: Store<JobPublicationDetailState>,
                 private coreStore: Store<CoreState>,
-                private jobAdvertisementCancelDialogService: JobAdvertisementCancelDialogService) {
+                private jobAdvertisementCancelDialogService: JobAdvertisementCancelDialogService,
+                private route: ActivatedRoute,
+                private router: Router) {
+        this.route.queryParamMap.subscribe((params) => this.token = params.get('token'));
```
[the remark this answers is not in the document]
Comment (Invizible) [reply src/main/webapp/app/dashboard/job-publication-detail/job-publication-detail.component.ts]: why should it be there? I'm not interested in @ Input() properties, so there is no need in OnInit

File: src/main/webapp/app/shared/auth/auth-jwt.service.ts (hunk 2 above)
Comment (cjpsycos) [reply src/main/webapp/app/shared/auth/auth-jwt.service.ts]: @pavax is it a problem?

File: src/main/webapp/app/dashboard/job-publication-detail/job-publication-detail.component.ts [hunk 4]
```diff
@@ -58,9 +67,17 @@ export class JobPublicationDetailComponent {
...
     }
 
-    showCancellationDialog(id: string) {
+    showCancellationDialog(id: string, token: string) {
         const onSubmit = (cancellationData) => this.store.dispatch(new SubmitCancellationAction(cancellationData));
-        this.jobAdvertisementCancelDialogService.open(id, onSubmit);
+        this.jobAdvertisementCancelDialogService.open(id, onSubmit, token);
+    }
+
+    copyJobAdvertisement(id: string, token: string) {
```
Comment (pavax) [inline src/main/webapp/app/dashboard/job-publication-detail/job-publication-detail.component.ts]: why do we copy the token and the id?

File: src/main/webapp/app/dashboard/job-publication-detail/job-publication-detail.component.ts [hunk 5]
```diff
@@ -58,9 +67,17 @@ export class JobPublicationDetailComponent {
...
+        const params = { jobPublicationId: id };
+        if (token) {
```
Comment (pavax) [inline src/main/webapp/app/dashboard/job-publication-detail/job-publication-detail.component.ts]: why do we need the token

File: src/main/webapp/app/home/tools/job-publication-tool/service/job-advertisement-resolver.service.ts [hunk 6]
```diff
@@ -5,23 +5,34 @@ import { JobAdvertisementService } from '../../../../shared/job-advertisement/jo
...
     constructor(private jobAdvertisementService: JobAdvertisementService,
                 private occupationPresentationService: OccupationPresentationService,
-                private translateService: TranslateService) {
+                private translateService: TranslateService,
+                private peaDashboardGuard: PeaDashboardGuard) {
     }
 
     resolve(route: ActivatedRouteSnapshot, state: RouterStateSnapshot): Observable<JobAdvertisement> {
         const jobPublicationId = route.paramMap.get('jobPublicationId');
+        const token = route.paramMap.get('token');
```
Comment (pavax) [inline src/main/webapp/app/home/tools/job-publication-tool/service/job-advertisement-resolver.service.ts]: I'm not sure wheter we need the token here as well?

**Pull request merged**